zipline/pipeline/graph.py

```diff
@@ -132,15 +141,28 @@ def initial_refcounts(self, initial_terms):
         nodes get one extra reference to ensure that they're still in the graph
         at the end of execution.
         """
-        refcounts = self.out_degree()
+        refcounts = self.graph.out_degree()
         for t in self.outputs.values():
             refcounts[t] += 1
 
         for t in initial_terms:
-            self.decref_dependencies(t, refcounts)
+            self._decref_recursive(t, refcounts, set())
 
         return refcounts
 
+    def _decref_recursive(self, term, refcounts, garbage):
+        """
+        Decrement terms recursivly to build the initial workspace.
```

Comment:
This might benefit from a description of when this should be used versus `decref_dependencies`.  I think the distinction is that this should be used when computing initial refcounts on the graph from a set of precomputed terms, whereas `decref_dependencies` should be used during computation to decrement a term that was just computed.